src/main/java/net/hypixel/nerdbot/curator/ChannelGroupCurator.java

```diff
@@ -0,0 +1,321 @@
+     */
+    public void sendGreenlitToChannel(TextChannel channel, ChannelGroup channelGroup, GreenlitMessage message) {
+        channel.sendMessageEmbeds(message.getEmbed().build()).queue(message1 -> {
+            message.setSuggestionTitle(message1.getId());
+            Database.getInstance().insertGreenlitMessage(message);
+            log("[" + channelGroup.getName() + "] [" + message.getId() + "] Greenlit message sent to " + channel.getName() + " and inserted into database");
+        });
+    }
+
+    /**
+     * Add an emoji to a message if it doesn't already exist
+     *
+     * @param group   The {@link ChannelGroup} to search through
+     * @param message The {@link Message} to add the emoji to
+     * @param emoji   The {@link Emoji} to add to the message
+     */
+    private void addEmojiIfMissing(ChannelGroup group, Message message, @NotNull Emoji emoji) {
+        if (message.getReaction(emoji) == null) {
+            log("[" + group.getName() + "] [" + message.getId() + "] No reaction found, adding one!");
+            message.addReaction(emoji).queue();
+        }
+    }
+
+    /**
+     * Create a new {@link GreenlitMessage} from a {@link Message}
+     *
+     * @param message  The {@link Message} to create the {@link GreenlitMessage} from
+     * @param positive The positive amount of {@link MessageReaction}
+     * @param negative The negative amount of {@link MessageReaction}
+     *
+     * @return The new {@link GreenlitMessage}
+     */
+    private GreenlitMessage createGreenlitMessage(ChannelGroup group, Message message, int positive, int negative) {
+        return GreenlitMessage.builder().userId(message.getAuthor().getId()).messageId(message.getId()).tags(getTags(message.getContentRaw())).suggestionTitle(Util.getFirstLine(message)).suggestionContent(message.getContentRaw()).suggestionDate(new Date(message.getTimeCreated().toInstant().toEpochMilli())).suggestionUrl(message.getJumpUrl()).agrees(positive).disagrees(negative).channelGroupName(group.getName()).build();
+    }
+
+    /**
+     * Get all tags that a message contains in the first line
+     *
+     * @param message The {@link Message} to get the tags from
+     *
+     * @return A list of tags
+     */
+    private List<String> getTags(String message) {
+        String firstLine = message.split("\n")[0];
+        Matcher matcher = Util.SUGGESTION_TITLE_REGEX.matcher(firstLine);
+        List<String> tags = new ArrayList<>();
+
+        while (matcher.find()) tags.add(matcher.group(1));
+
+        return tags;
+    }
+
+    /**
+     * Remove all reactions from a message by a user or bot
+     *
+     * @param message  The {@link Message} to remove the reactions from
+     * @param reaction The {@link MessageReaction} to check and remove from
+     */
+    private int discountBotAndUserReactions(ChannelGroup group, Message message, MessageReaction reaction) {
+        AtomicInteger total = new AtomicInteger(reaction.getCount());
```

Comment:
why am i atomic?

Reply:
It's being used in the lambda but I've just largely changed this method, so there's no need for it anymore